js/openstates.js

```diff
@@ -0,0 +1,63 @@
+/*
+This file moves the OpenStates API calls into their own module, and caches responses for speed on second request.
+*/
+
+var API_KEY = "9e3e71730ae34e1ebbf4dd0e1c346c07";
+var prefix = "http://openstates.org/api/v1/";
+
+var $ = require("jquery");
+
+var cache = {
+  states: {},
+  committees: {},
+  people: {}
+}
+
+var getState = function(state) {
+  if (!cache.states[state]) {
+    cache.states[state] = $.ajax({
```

Comment:
There's actually a lot of stuff here that we could probably consolidate--on each request, we specify the same `dataType`, and usually send in the API key, so we could probably write a function to eliminate that repetition.

The cache works by storing the promises, and since it expects those there's a chance to do further processing here. For example, we could pre-process results by writing this code instead:

```js
if (!cache.states[state]) {
  cache.states[state] = $.Deferred(function(deferred) { //create a promise
    $.ajax({ /* ...request options */ }).done(function(data) {
      //process committees in some way...
      data.committees.forEach(function(c) {
        //...
      });
      // return the data via the outer promise for the cache
      deferred.resolve(data);
    }).promise();
}